src/viadot/orchestration/prefect/tasks/mediatool.py

```diff
@@ -0,0 +1,165 @@
+"""'mediatool.py'."""
+
+from typing import Any
+
+import pandas as pd
+from prefect import get_run_logger, task
+
+from viadot.orchestration.prefect.exceptions import MissingSourceCredentialsError
+from viadot.orchestration.prefect.utils import get_credentials
+from viadot.sources import Mediatool
+from viadot.utils import join_dfs
+
+
+@task(retries=3, log_prints=True, retry_delay_seconds=10, timeout_seconds=60 * 60)
+def mediatool_to_df(
+    credentials: dict[str, Any] | None = None,
+    config_key: str | None = None,
+    azure_key_vault_secret: str | None = None,
+    organization_ids: list[str] | None = None,
+    media_entries_columns: list[str] | None = None,
+) -> pd.DataFrame:
+    """Task to download data from Mediatool API.
+
+    Data from different endpoints are retrieved and combined. A final object is created
+    containing data for all organizations from the list.
+
+    Args:
+        credentials (dict[str, Any], optional): Mediatool credentials as a dictionary.
+            Defaults to None.
+        config_key (str, optional): The key in the viadot config holding relevant
+            credentials. Defaults to None.
+        azure_key_vault_secret (str, optional): The name of the Azure Key Vault secret
+            where credentials are stored. Defaults to None.
+        organization_ids (list[str], optional): List of organization IDs.
+            Defaults to None.
+        media_entries_columns (list[str], optional): Columns to get from media entries.
+            Defaults to None.
+
+    Raises:
+        ValueError: 'organization_ids' argument is None, and one is mandatory.
+        ValueError: One 'organization_id' is not in organizations.
+
+    Returns:
+        pd.DataFrame: The response data as a Pandas Data Frame.
+    """
+    logger = get_run_logger()
+
+    if not (azure_key_vault_secret or config_key or credentials):
+        raise MissingSourceCredentialsError
+
+    if not config_key:
+        credentials = credentials or get_credentials(azure_key_vault_secret)
+
+    mediatool = Mediatool(
+        credentials=credentials,
+        config_key=config_key,
+    )
+    # first method ORGANIZATIONS
+    method = "organizations"
+    organizations_data = mediatool.api_connection(get_data_from=method)
+    df_organizations = mediatool.to_df(data=organizations_data, column_suffix=method)
```

Comment:
There is an existing standard term for this https://blog.postman.com/what-is-an-api-endpoint/, so (applying the other feedback as well) this code could be simplified to

```suggestion
    df_organizations = mediatool.to_df(endpoint="organizations")
```

Reply:
I think this is an issue for a backlog.

Reply:
Then pls create an issue for it and link it here so we can close this

Reply:
#1060